src/Hypsography/Hypsography.jl

```diff
@@ -77,10 +77,12 @@ end
 Locate the levels by linear interpolation between the surface field and the top
 of the domain, using the method of [GalChen1975](@cite).
 """
-struct LinearAdaption{F <: Fields.Field} <: HypsographyAdaption
+struct LinearAdaption{F} <: HypsographyAdaption
     surface::F
 end
 
+strip_field(::LinearAdaption) = LinearAdaption(nothing)
```

Comment:
Shouldn't this be `strip_space`? And it should keep the field values

Reply:
There is no need to keep the field values in this function because they are passed in separately through `ref_z_to_physical_z`. I think it is best to avoid passing duplicate field values through here. In general, the design of this section should probably be refactored to split the surface elevations from the adaption specification.

Reply:
I have redesigned this to avoid passing the field values separately, instead using a lazy broadcast to pass them through the adaption. Keeping everything in the adaption seems more in line with the original intention of this data structure, and the new pattern is also more generalizable.